op-service/backoff/operation.go

```diff
@@ -42,3 +43,35 @@ func Do(maxAttempts int, strategy Strategy, op Operation) error {
 		time.Sleep(strategy.Duration(attempt - 1))
 	}
 }
+
+func DoCtx(ctx context.Context, maxAttempts int, strategy Strategy, op Operation) error {
```

Comment:
should `Do` just be a call to `DoCtx` with `context.Background` as the supplied parameter?

Reply:
Yeah good call, I'll update.